test/freenet/client/filter/GIFFilterTest.java

```diff
@@ -117,17 +117,16 @@ public void testFilterPairs() throws IOException {
 
     @Test
     public void testReject() throws IOException {
+        ContentDataFilter filter = new GIFFilter();
```

Comment:
I think this declaration was totally fine where it was, you know, close to its use.

Reply:
Moved back.
Initially tried remove the instantiation from the loop, but this is correct only if `GIFFilter()` has no any internal state.